### ISSUE TITLE
utils.TypeConverters.register with Long

### ISSUE THREAD

**Issue by danishin** [tminglei/slick-pg#235]
Hello, I'm trying to use Joda's `DateTime` in my `composite` by converting it into `Long`, which is eventually stored as `BIGINT` in postgres.

However, the following code throws exception at runtime.
```
utils.TypeConverters.register[DateTime, Long](_.getMillis)
utils.TypeConverters.register[Long, DateTime](new DateTime(_))
```
with stacktrace:
> ...
   Caused by: java.lang.IllegalArgumentException: Converter NOT FOUND for 'String' => 'org.joda.time.DateTime'
  ...

It seems using `String` in place of `Long` does work and I've found out that all of the default `TypeConverters` `register`ed in slick-pg are in fact conversions from & to `String` exclusively.

Perhaps, I'm missing something here. 
What does `TypeConverters` do exactly? Is it `TypeMapper` for `composite` type only?

Basically, I just want to use `DateTime` inside composite :smile:  Thanks for your help!

**Comment by danishin** [comment]
Using `TIMESTAMPTZ` as composite field type and the following works:
```
utils.TypeConverters.register[DateTime, String](_.toString)
utils.TypeConverters.register[String, DateTime](a => new DateTime(a))
```

It seems like the conversion between `String` -> `TIMESTAMPTZ` is bridged by `java.sql.TimeStamp` and what's being returned by `datetime.toString` and `timestamp.toString` (as defined by slick-pg internally) are compatible so everything just works magically.

Thanks for a great library @tminglei !

**Comment by tminglei** [comment]
Glad to hear that! _You're welcome! :-)_
